### ISSUE TITLE
Connection/Player Count Limiting

### ISSUE THREAD

**Issue by lodicolo** [AscensionGameDev/Intersect-Engine#77]
**Depends on**: #63 

**Is your feature request related to a problem? Please describe.**
The server does not have a connection limit, nor a player limit, and will accept connections and players up until it runs out of resources.

**Describe the solution you'd like**
Server should:
- Have the ability to limit concurrent connections (not logged in, "Connections")
  - Should be configurable
  - Should immediately disconnect clients that request a connection beyond this threshold with `NetworkStatus.ServerFull`
- Have the ability to limit concurrent active connections (logged in, "Players")
  - Should be configurable
  - Should reject logins but not terminate the connection (and in a future separate issue can be expanded upon for login queues)

**Additional context**
See #63 for server status details request
